brave/src/main/java/brave/Tracing.java

```diff
@@ -97,16 +97,25 @@ public final Clock clock(TraceContext context) {
 
   abstract public ErrorParser errorParser();
 
-  // volatile for visibility on get. writes guarded by Tracing.class
-  static volatile Tracing current = null;
+  // volatile for get visibility. writes guarded by Tracing.class. Object to ensure unloadable
+  static volatile Object current = null;
+
+  /**
+   * Returns the most recently created tracing component iff it hasn't been closed. null otherwise.
+   *
+   * <p>This object should not be cached.
+   */
+  @Nullable public static Tracing current() {
+    return (Tracing) current;
+  }
 
   /**
    * Returns the most recently created tracer if its component hasn't been closed. null otherwise.
    *
    * <p>This object should not be cached.
    */
   @Nullable public static Tracer currentTracer() {
-    Tracing tracing = current;
+    Tracing tracing = (Tracing) current;
```

Comment:
in case forgot the magic java rules like I did.. no, cast won't NPE. Also, we have tests too.